pandas/core/ops/__init__.py

```diff
@@ -353,6 +354,70 @@ def fill_binop(left, right, fill_value):
 # Dispatch logic
 
 
+def operate_blockwise(left, right, array_op):
+    assert right._indexed_same(left)
+
+    res_blks: List["Block"] = []
+    rmgr = right._data
+    for n, blk in enumerate(left._data.blocks):
+        locs = blk.mgr_locs
+
+        blk_vals = blk.values
+
+        if not isinstance(blk_vals, np.ndarray):
+            # 1D EA
+            assert len(locs) == 1, locs
```

Comment:
might be more clear to make for example this section (368 to 374) as a function

so the structure of what you are doing is more clear.